### PR TITLE
Use unique iframe names based on unique domain

### PR DESCRIPTION
**Pull request by jridgewell**
Browsers are weird. If an iframe has a `name` attribute, it’s window
will be cached. When you click a link, then go back, it will insert the
**same** src window into the iframe regardless of a different `src`.

Well, that hurts us since we generate a unique domain to load 3p
frames. We’ll continue to use an old domain’s window, even though the
src points to a new domain. Worse, when we post messages to the 3p
frame, we will send it with an origin parsed from the `src` attribute
even though that is not the real URL of the frame’s window.

Fixes https://github.com/ampproject/amphtml/issues/2943.

### CODE REVIEW

**Comment by cramforce** [comment]
Thanks! Please add a test.

**Comment by jridgewell** [comment]
Added.

File: test/functional/test-3p-frame.js [hunk 1]
```diff
@@ -238,4 +245,31 @@ describe('3p-frame', () => {
...
+        width: 100,
+        height: 200,
+      };
+    };
+
+    const name = getIframe(window, div).name;
+    resetServiceForTesting(window, 'bootstrapBaseUrl');
+    resetCountForTesting();
+    const newName = getIframe(window, div).name;
+    expect(newName).not.to.equal(name);
```
Comment (cramforce) [inline test/functional/test-3p-frame.js]: It'd be really good to hard code the actual name string in a test.

Comment (jridgewell) [inline test/functional/test-3p-frame.js]: Added.

File: test/functional/test-3p-frame.js [hunk 2]
```diff
@@ -238,4 +245,33 @@ describe('3p-frame', () => {
     };
     expect(getSubDomain(fakeWin)).to.equal('d-5670');
   });
+
+  it.only('uses a unique name based on domain', () => {
```
Comment (cramforce) [inline test/functional/test-3p-frame.js]: `.only`

**Comment by cramforce** [comment]
LGTM

File: test/functional/test-3p-frame.js (hunk 2 above)
Comment (jridgewell) [inline test/functional/test-3p-frame.js]: 😔

**Pull request merged**